sources/api/migration/migrator/src/args.rs

```diff
@@ -33,18 +33,18 @@ fn usage_msg<S: AsRef<str>>(msg: S) -> ! {
 }
 
 /// Stores user-supplied arguments.
-pub(crate) struct Args {
```

Comment:
This is exactly the reason I don't like this pattern - everything has to be `pub`.  Until/unless Rust supports `main.rs` being part of the same crate, this really bugs me, and it implies a lot more restrictions on what you can do in the library, since you have to be concerned about consumers.

Why do we want this change, at a high level?  You had the test running before, without this reorganization...  I don't like the word "unit" versus "integration" as a reason for disrupting code organization that actually impacts our developers :/

Reply:
> Why do we want this change, at a high level?

https://github.com/bottlerocket-os/bottlerocket/pull/930#discussion_r443870244

I will push a change to this PR in which the only reorganization required (I think) is to move `run` to `src/run.rs` and make it `pub(crate` and to add a file `src/run_test.rs` to house the test code.

This should prove less disruptive.

Reply:
Edit: Unfortunately, I'm now realizing everything would have to move to `run.rs` except for the main function itself. This will be just as disruptive to the git history as the lib.rs version, but it will not expose anything as `public` and will not expose a `lib.rs`. I'll open it in a bit.

On a side note: I have the personal takeaway. Never, ever, define anything in `main.rs` except for the `main` function itself. No matter how small you think your program will remain. Anyway, that will be personal mantra when starting Rust `bin` projects from now on! It's not so different from C++, you can't test anything that's defined in `main.cpp`.

Reply:
I'm definitely missing something.  If `main.rs` can call things from `run.rs` as part of the same crate (no `lib` crate) then why can't we have a `test.rs` that's part of the `main` crate and testing the things you want to test?

Reply:
We can, but test.rs cannot call anything defined in main.rs. main.rs is not a module that you can `use`.

Reply:
The pushed version moves everything to `run.rs` and tests it in `run_test.rs` which is conditionally compiled for cfg test only.

I added a commit where I did a `git mv main.rs run.rs` and I think it does help retain the ability to follow the lineage of the code.

Reply:
If `main` can call `run` without `pub`, then they're in the same crate and are accessible to each other.  Why can't `test.rs` be a `#[cfg(test)] mod test` declared in `main.rs` and run your unit test?  I don't see any difference between "main + run + run_test" and "main + test" - they're both running a unit test from the "main" crate.

Reply:
It is. I think we're typing past each other. What I have no pushed and is ready for your review is, I think, exactly what you are saying (and I was trying to say).

Reply:
> It is. I think we're typing past each other. What I have now pushed and is ready for your review is, I think, exactly what you are saying (and I was trying to say).

Or... maybe I'm missing something, if what you're say is that I could `use crate::main::run` in `src/run_test.rs` then that is something I am assuming I can not do. I'll see if that's possible.

Reply:
Ah, I see that my assumption was wrong; this works

`src/main.rs`

```rust
#[cfg(test)]
mod main_test;

fn main() {
    println!("Hello, world!");
}

pub(crate) fn something() -> i32 {
    0
}
```

```rust
#[cfg(test)]
mod test {
    use crate::something;
    #[test]
    fn test_something() {
        assert_eq!(something(), 0);
    }
}
```

I will use this approach. Ignore re-review request until further notice.

Reply:
In short, I'd like to get rid of the `run` files because I don't think the further separation serves a purpose - I don't think it's needed for visibility reasons.  I think we can have `#[cfg(test)] mod test;` in `main.rs`, and have a `test.rs` with the test in it, which should be able to access things from `main` because they're in the same "main" crate.  Does that make sense?